__tests__/p2sh/tx-proposal-new-token.test.js

```diff
@@ -0,0 +1,438 @@
+      .send({
+        name: 'My Custom Token',
+        symbol: 'MCT',
+        amount: 1,
+        create_mint: false,
+        create_melt: true,
+        melt_authority_address: '',
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(400);
+    expect(response.body.success).toBe(false);
+  });
+
+  it('should return 200 with a valid body with a melt authority selecting an external authority address', async () => {
+    const meltAuthorityAddress = TestUtils.addresses[1];
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/create-token')
+      .send({
+        name: 'My Custom Token',
+        symbol: 'MCT',
+        amount: 1,
+        create_mint: false,
+        create_melt: true,
+        melt_authority_address: meltAuthorityAddress,
+        allow_external_melt_authority_address: true,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.txHex).toBeDefined();
+    const tx = hathorLib.helpersUtils.createTxFromHex(response.body.txHex, new hathorLib.Network('testnet'));
+    expect(tx.outputs.map(o => o.decodedScript.address.base58))
+      .toEqual(expect.arrayContaining([TestUtils.multisigAddresses[1], meltAuthorityAddress]));
+    expect(tx.outputs).toHaveLength(3);
+    const authorityOutputs = tx.outputs.filter(o => TOKEN_DATA.isAuthorityToken(o.tokenData));
+    expect(authorityOutputs).toHaveLength(1);
+    expect(authorityOutputs[0].value).toBe(AUTHORITY_VALUE.MELT);
+  });
+
+  it('should return 200 with a valid body with a mint and melt authority selecting an authority address', async () => {
+    const mintAuthorityAddress = TestUtils.multisigAddresses[2];
+    const meltAuthorityAddress = TestUtils.multisigAddresses[3];
+    const response = await TestUtils.request
+      .post('/wallet/p2sh/tx-proposal/create-token')
+      .send({
+        name: 'My Custom Token',
+        symbol: 'MCT',
+        amount: 1,
+        create_mint: true,
+        mint_authority_address: mintAuthorityAddress,
+        create_melt: true,
+        melt_authority_address: meltAuthorityAddress,
+      })
+      .set({ 'x-wallet-id': walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.success).toBe(true);
+    expect(response.body.txHex).toBeDefined();
+    const tx = hathorLib.helpersUtils.createTxFromHex(response.body.txHex, new hathorLib.Network('testnet'));
+    expect(tx.outputs.map(o => o.decodedScript.address.base58))
+      .toEqual(expect.arrayContaining([
+        TestUtils.multisigAddresses[1],
+        mintAuthorityAddress,
+        meltAuthorityAddress
+      ]));
+    expect(tx.outputs).toHaveLength(4);
+    const authorityOutputs = tx.outputs.filter(o => TOKEN_DATA.isAuthorityToken(o.tokenData));
+    expect(authorityOutputs).toHaveLength(2);
+    expect(authorityOutputs[0].value).toBe(AUTHORITY_VALUE.MINT);
```

Comment:
Can we be sure the order will always be `[mint, melt]`?
If it changes order in a non-deterministic way, we could have intermittent test failures here. Otherwise, I think it's fine like it is.

Reply:
Yes, yes. This is the order.